chainer/testing/distribution_test.py

```diff
@@ -0,0 +1,435 @@
+import unittest
+
+import chainer
+from chainer.backends import cuda
+import numpy
+
+try:
+    from chainer.testing import attr
+    _error = attr.get_error()
+except ImportError as e:
+    _error = e
+
+
+def check_available():
+    if _error is not None:
+        raise RuntimeError('''\
+{} is not available.
+
+Reason: {}: {}'''.format(__name__, type(_error).__name__, _error))
+
+
+def distribution_unittest(dist, scipy_dist_name, params_init, sample_for_test,
+                          tests=set(), continuous=True, support="real",
+                          event_shape=(), scipy_onebyone=False):
```

Comment:
How about making it just a base class instead of a class decorator? Options can be provided as methods or (class) attributes of each child class. Conditional tests with `tests` may be provided by a method decorator that checks the `tests` attribute of the child class (btw, I think the name `tests` should be more explicit, like `test_targets`).